test/envtest/ingress_test.go

```diff
@@ -27,6 +27,15 @@ import (
 	"github.com/kong/kubernetes-ingress-controller/v3/test/helpers"
 )
 
+// configDumpResponse mirrors the diagnostics.configDumpResponse struct, which isn't published.
+// It's replicated here since some envtests use the config dump endpoints as a hack to extract the config for
+// inspection.
+
+type configDumpResponse struct {
+	ConfigHash string       `json:"hash"`
+	Config     file.Content `json:"config"`
+}
```

Comment:
I think it wouldn't be a bad practice to make the response types public from `pkg/diagnostics` package and avoid this duplication. Can be handled in a separate PR though.

Reply:
I'd prefer to keep them internal unless we have good reason to (we've already got enough fun with annoying dependency loops trying to reuse stuff in fun places). This seemed like an odd enough corner of the codebase (tests only, and a particular weird case where a small set of envtests want to interact with non-Kubernetes resources) that it can use a shim that's hopefully unlikely to change further for the purposes of what we need--the change from root to key is hopefully a one-off change, where any future changes to the diag APIs is adding new keys or endpoints, not moving the `config` key elsewhere.